astropath/bayesian.py

```diff
@@ -0,0 +1,190 @@
+"""Methods related to Bayesian association analysis"""
+
+import numpy as np
+
+import copy
+
+from astropy import units
+from astropy.coordinates import SkyCoord
+
+from IPython import embed
+
+
+def raw_prior_Oi(Pchance, Sigma_m, method, half_light=None):
+    """
+    Raw prior for a given set of Pchance values and/or n(m)
+
+    Args:
+        Pchance (float or np.ndarray):
+            Chance probability
+        Sigma_m (float or np.ndarray):
+            Number density of sources on the sky brighter than m
+        method (str):
+            inverse :: Assign inverse to Sigma_m
+            inverse1 :: Assign inverse to Sigma_m * half_light
+            inverse2 :: Assign inverse to Sigma_m * half_light**2
+            orig_inverse :: Assign inverse to P_chance
+            identical :: All the same
+            linear :: 1-Pchance (not recommended)
+        half_light (float or np.ndarray, optional):
+            Angular size of the galaxy
+            Only required for several methods
+
+    Returns:
+        float or np.ndarray:
+
+    """
+    if method == 'linear':
+        return 1 - Pchance
+    elif method == 'orig_inverse':
+        return 1. / Pchance
+    elif method == 'inverse':
+        return 1. / Sigma_m
+    elif method == 'inverse1':
+        return 1. / Sigma_m / half_light
```

Comment:
Should this be 1. / Sigma_m * half_light?

Reply:
no, `half_light` needs to be in the denominator.
as it was with P_chance

Reply:
I am also confused by this. I thought it was to be Sigma_m * half_light and Sigma_m * half_light**2. Only in this case units would work out for the second one, I think.

Reply:
Actually, no. This is right. 1. / Sigma_m / half_light is same as 1 / (Sigma_m * half_light).